public/src/components/Main.tsx

```diff
@@ -145,22 +154,51 @@ export class Main extends React.Component<MainProps, MainState> {
         }
     }
 
-    retryStream = (func: Function) => {
-        let newcounter = this.state.networkTimeOutCounter;
-        let newTimeout = this.state.networkTimeOut + Math.floor(newcounter/3) * this.state.networkTimeOut;
-
-        if (newcounter == 2) {
-            const timeOut = this.state.networkTimeOut;
-            showErrorNotif("Unable to connect to server. Please check your internet connection. Retrying in " + (timeOut/1000) + "s", "Network error");
-            setTimeout(func, timeOut);
+    retryStream = (func: Function, flag: string) => {
```

Comment:
1. Create a function ```getStreamCounter(flag: string)```  which returns the appropriate counter.
2. Create a function ```setStreamCounter(flag: string, value: number)``` which sets the appropriate counter
3. No duplicate code. Don't rewrite anything.
4. What does ```forErrorNotifs()``` supposed to mean? Conveys 0 information about what the function does.